crates/matrix-sdk/src/client/mod.rs

```diff
@@ -582,8 +587,9 @@ impl Client {
     ///     // You can omit any or all arguments after the first.
     /// });
     ///
-    /// // Registering a temporary event handler:
-    /// let handle = client.add_event_handler(|ev: SyncRoomMessageEvent| async move {
+    /// // Registering a temporary event handler with custom context:
+    /// client.add_event_handler_context("Custom context type is &str here");
```

Comment:
I think using `&str` as `Ctx` is pretty far from typical usage, and also using it for a temporary handler seems especially questionable, because it will sit around after the handler is removed. Can you instead define an empty struct and add another `add_event_handler` call (before this one would be a good place I think).

Reply:
Thanks, indeed, fixed. 

As of the issue, I am not the creator, but as it stated as alternative that an example in the README would have been enough too, I think it is about discoverability of the use of `Ctx`. I would consider it visible, but I am not the issue creator, so I wrote "supposed to close" :D Not sure